src/eradius_server.erl

```diff
@@ -204,39 +199,35 @@ do_radius(ServerPid, ReqKey, Handler = {HandlerMod, _}, NasProp, {udp, Socket, F
             TS2 = eradius_metrics:timestamp(milli_seconds),
             inc_counter(Cmds, NasProp, TS2 - TS1),
             gen_udp:send(Socket, FromIP, FromPort, EncReply),
-            case application:get_env(eradius, resend_timeout, 2000) of
-                0 -> 
-                    ServerPid ! {free, ReqKey};
-                ResendTimeout ->
-                    ServerPid ! {replied, ReqKey, self()},
-                    wait_resend_init(ServerPid, ReqKey, FromIP, FromPort, EncReply, ResendTimeout, ?RESEND_RETRIES)
+            ResendTimeout = application:get_env(eradius, resend_timeout, 2000),
+            if ResendTimeout > 0 ->
```

Comment:
You need a guard to ensure ResendTimeout is integer (if resend_timeout wasn't validated to be integer somewhere else already), the simple 'greater that' works for arbitrary terms, e.g.:

```
1> a > 0.
true
```